File: Blueprints/BlueprintDefinitions/Msbuild-NETCore_2_0/StepFunctionsHelloWorld-FSharp/template/test/BlueprintBaseName.1.Tests/FunctionTest.fs
namespace BlueprintBaseName._1.Tests


open Xunit
open Amazon.Lambda.TestUtilities

open BlueprintBaseName._1


module FunctionTest =
    [<Fact>]
    let ``Test Greeting Function``() =
        let context = TestLambdaContext()
        let state =
            StepFunctionTasks.Greeting
                (StepFunctionTasks.State(Name = "MyStepFunctions"))
                context

        Assert.Equal(5, state.WaitInSeconds)
        Assert.Equal("Hello MyStepFunctions", state.Message)

    [<EntryPoint>]
    let main _ = 0

File: Blueprints/BlueprintDefinitions/vs2026/StepFunctionsHelloWorld/template/test/BlueprintBaseName.1.Tests/FunctionTest.cs
using Xunit;
using Amazon.Lambda.Core;
using Amazon.Lambda.TestUtilities;

namespace BlueprintBaseName._1.Tests;

public class FunctionTest
{
    public FunctionTest()
    {
    }

    [Fact]
    public void TestGreeting()
    {
        TestLambdaContext context = new TestLambdaContext();

        StepFunctionTasks functions = new StepFunctionTasks();

        var state = new State
        {
            Name = "MyStepFunctions"
        };


        state = functions.Greeting(state, context);

        Assert.Equal(5, state.WaitInSeconds);
        Assert.Equal("Hello MyStepFunctions", state.Message);
    }
}